examples/plot_classification_example.py

```diff
@@ -0,0 +1,247 @@
+rng = np.random.RandomState(0)
+X, y = dataset.data, dataset.target
+spurious_feat = X[:, 2] + X[:, 3]
+spurious_feat += rng.normal(size=X.shape[0], scale=np.std(spurious_feat) / 2)
+X = np.hstack([X, spurious_feat.reshape(-1, 1)])
+
+dataset.feature_names = dataset.feature_names + ["spurious_feat"]
+
+
+############################################################################
+# Measure variable importance
+# --------------------------------------------------------------------------
+# Since all three methods compute variable importance as a loss difference, they
+# require a K-fold cross-fitting. Computing the importance for each fold is
+# embarassingly parallel, so we use joblib to parallelize the computation.
+def run_one_fold(X, y, model, train_index, test_index, vim_name="CPI", groups=None):
+    model_c = clone(model)
+    model_c.fit(X[train_index], y[train_index])
+    y_pred = model_c.predict(X[test_index])
+
+    if isinstance(model_c, LogisticRegressionCV):
+        method = "predict_proba"
+        loss = log_loss
+        model_name = "LogReg"
+    else:
+        method = "decision_function"
+        loss = hinge_loss
+        model_name = "SVC"
+    # else:
+    #     method = "predict_proba"
+    #     loss = log_loss
+    #     model_name = "RF"
+
+    if vim_name == "CPI":
+        vim = CPI(
+            estimator=model_c,
+            imputation_model_continuous=RidgeCV(alphas=np.logspace(-3, 3, 10)),
+            n_permutations=50,
+            random_state=0,
+            method=method,
+            loss=loss,
+        )
+    elif vim_name == "LOCO":
+        vim = LOCO(
+            estimator=model_c,
+            loss=loss,
+            method=method,
+        )
+    elif vim_name == "PermutationImportance":
+        vim = PermutationImportance(
+            estimator=model_c,
+            n_permutations=50,
+            random_state=0,
+            method=method,
+            loss=loss,
+        )
+
+    vim.fit(X[train_index], y[train_index], groups=groups)
+    importance = vim.importance(X[test_index], y[test_index])["importance"]
+
+    return pd.DataFrame(
+        {
+            "feature": groups.keys(),
+            "importance": importance,
+            "vim": vim_name,
+            "model": model_name,
+            "score": balanced_accuracy_score(y_true=y[test_index], y_pred=y_pred),
+        }
+    )
+
+
+models = [
+    LogisticRegressionCV(Cs=np.logspace(-3, 3, 10)),
+    GridSearchCV(LinearSVC(), {"C": np.logspace(-3, 3, 10)}),
+]
+cv = KFold(n_splits=5, shuffle=True, random_state=0)
+groups = {ft: i for i, ft in enumerate(dataset.feature_names)}
+out_list = Parallel(n_jobs=5)(
+    delayed(run_one_fold)(
+        X, y, model, train_index, test_index, vim_name=vim_name, groups=groups
+    )
+    for train_index, test_index in cv.split(X)
+    for model in models
+    for vim_name in ["CPI", "LOCO", "PermutationImportance"]
+)
+df = pd.concat(out_list)
+
+
+def compute_pval(df, threshold=0.05):
```

Comment:
Would it be possible to avoid such functions and directly write loops in the examples ?.  This makes the examples harder to understand

Reply:
I wrote a function because the code is used twice (once for single variables and once for groups). Would you rather have the loop duplicated?

Reply:
I see. I find this a bit frustrating for users because the interaction with the classes is burried in functions.
I'll try to make a suggestion upon my next review.

Reply:
I am not sure to see what you are referring to. Is it the `compute_pval` function, which only computes a test and does data frame manipulations? Or is it the `run_one_fold`? 
IMO the latter is useful to show how to parallelize the computation of importance over K-folds.

Reply:
In my opinion, you should remove this function and create 2 examples: one for grouping variables and one without it. You can make references between them to include the comparison.

Reply:
I get the point. (and I think that this pattern is also used in sklearn, so...)